### PR TITLE
I386 build fixes

### PR DESCRIPTION
**Pull request by bsdjhb**
Note the OFED one is partially due to the fact I'm just doing a plain 'make buildkernel' rather than using cheribuild (there is no cheribsd-i386 target).  (Though I do hope to eventually enable OFED on at least non-CHERI architectures)

### CODE REVIEW

**Review by brooksdavis — APPROVED** [review]
Looks good. A couple comments for context.

File: sys/sys/systm.h [hunk 1]
```diff
@@ -531,7 +531,7 @@ int64_t	fuword64(volatile const void * __capability base);
 #if __has_feature(capabilities)
 int	fuecap(volatile const void * __capability base, intcap_t *val);
 #else
-#define	fuecap		fueword
+#define	fuecap(base, val)		fueword((base), (long *)(val))
 #endif
```
[the remark this answers is not in the document]
Comment (bsdjhb) [reply sys/sys/systm.h:535]: Yeah, I would like that as well.  Maybe I will try to do that upstream.

File: sys/i386/i386/exec_machdep.c [hunk 2]
```diff
@@ -653,9 +653,9 @@ osigreturn(struct thread *td, struct osigreturn_args *uap)
 int
 freebsd4_sigreturn(struct thread *td, struct freebsd4_sigreturn_args *uap)
 {
-	struct ucontext4 uc;
+	struct freebsd4_ucontext uc;
```
[the remark this answers is not in the document]
Comment (bsdjhb) [reply sys/i386/i386/exec_machdep.c:656]: Ok.  Maybe I will rename mcontext4 upstream and let it loop back in.

**Comment by bsdjhb** [comment]
CI X was a timeout, so I think a red herring.

**Pull request merged**